### DISCUSSION TITLE
Scanning the qr code with the phone(ios) that is displayed on the modal window - Web3Modal

### DISCUSSION THREAD

**Discussion by SerjioValentes** [WalletConnect/walletconnect-monorepo#2453] [Developer Support] [answered]
- connecting web3Modal
- open modal on desktop
- scaning qr code by phone(ios)
- auto opening in mobile device metamask
Issue:
-  after scanning i dosn't see sign on mobile metamask app

And this happened with your demo apps too
I saw a push method in alfa version - is it what i have to use to connect sign desktop and mobile?

**Comment by glitch-txs** [comment]
Hi, if you're running web3modal v2, MetaMask doesn't have support for it yet

**Reply by SerjioValentes** [reply]
> Hi, if you're running web3modal v2, MetaMask doesn't have support for it yet

And i have some another options to make modal with wallet connection? 
I mean if i'll use methods from wallet connect to make modal (without web3Modal), is it possible to connect metamask qr code with scanning by phone?

**Reply by glitch-txs** [reply]
Yes there are other options, can you share a snippet of your code and dependencies?

**Reply by SerjioValentes** [reply]
Package.json
"@web3modal/ethereum": "^2.3.7",
"@web3modal/react": "^2.3.7",

autoConnect: true, 
connectors: w3mConnectors({ projectId, version: 2, chains }), 
publicClient, 
webSocketPublicClient 
});
`
<WagmiConfig config={config}>
<Components />
</WagmiConfig>
<Web3Modal projectId={projectId} ethereumClient={ethereumClient} />
`

**Reply by glitch-txs** [reply]
Good, on connectors set the version flag to 1.
This will display WalletConnect v1, remember it will stop working on June 28 so you have to switch back to v2 before that date, MetaMask should support v2 by then as well

**Reply by SerjioValentes** [reply]
Yes, but v1 will end support at the end of June
Thats why i wanna use v2 ...

**Reply by glitch-txs** [reply, marked as answer]
MetaMask will support v2 soon, you can either wait or use v1 and then switch to v2
